src/lib.rs

```diff
@@ -276,22 +276,18 @@ impl OciDir {
     }
 
     #[cfg(feature = "zstd")]
-    /// Create a tar output stream, backed by a zstd compressed blob
+    /// Create a writer for a new zstd+tar blob; the contents
+    /// are not parsed, but are expected to be a tarball.
     ///
     /// This method is only available when the `zstd` feature is enabled.
-    pub fn create_layer_zstd(
-        &self,
-        compression_level: Option<i32>,
-    ) -> Result<tar::Builder<ZstdLayerWriter>> {
-        Ok(tar::Builder::new(ZstdLayerWriter::new(
-            &self.dir,
-            compression_level,
-        )?))
+    pub fn create_layer_zstd(&self, compression_level: Option<i32>) -> Result<ZstdLayerWriter> {
+        Ok(ZstdLayerWriter::new(&self.dir, compression_level)?)
```

Comment:
clippy with `--all-features` says we should remove `Ok` and `?` and just return:
```
ZstdLayerWriter::new(&self.dir, compression_level)
```
Same suggestion below in `create_layer_zstd_multithread`